Assets/Scripts/Data Model/Energy/EnergyPointSubEntity.cs

```diff
@@ -1,190 +1,192 @@
 		{
-			if (sourcePolygon.HasDatabaseID())
-				return sourcePolygon.GetDatabaseID().ToString();
-			else
-				return BatchRequest.FormatCallIDReference(sourcePolygon.Entity.creationBatchCallID);
+			connections = new List<Connection>();
+			sourcePolygon = sourcepoly;
 		}
-		return base.GetDataBaseOrBatchIDReference();
-	}
 
-	public override int GetPersistentID()
-    {
-        if (Entity.Layer.editingType == AbstractLayer.EditingType.SourcePolygonPoint)
-            return sourcePolygon.GetPersistentID();
-        return base.GetPersistentID();
-    }
+		public EnergyPointSubEntity(Entity entity, SubEntityObject geometry, int databaseID) : base(entity, geometry, databaseID)
+		{
+			if (Entity.Layer.editingType != AbstractLayer.EditingType.SourcePolygonPoint)
+				LayerManager.AddEnergySubEntityReference(databaseID, this);
+			connections = new List<Connection>();
+		}
 
-	public override void SubmitNew(BatchRequest batch)
-	{
-		base.SubmitNew(batch);
-	}
+		public override void SetDatabaseID(int databaseID)
+		{
+			if (Entity.Layer.editingType != AbstractLayer.EditingType.SourcePolygonPoint)
+				LayerManager.RemoveEnergySubEntityReference(databaseID);
+			this.databaseID = databaseID;
+			if (Entity.Layer.editingType != AbstractLayer.EditingType.SourcePolygonPoint)
+				LayerManager.AddEnergySubEntityReference(databaseID, this);
+		}
 
-	public override void SubmitDelete(BatchRequest batch)
-	{
-		base.SubmitDelete(batch);
+		/// <summary>
+		/// Called on points being placed to check if they can be connected to a certain endpoint
+		/// </summary>
+		public bool CanConnectToEnergySubEntity(EnergyPointSubEntity cableOrigin)
+		{
+			//Sources can't connect to sources
+			if ((Entity.Layer.editingType == AbstractLayer.EditingType.SourcePoint || Entity.Layer.editingType == AbstractLayer.EditingType.SourcePolygonPoint) &&
+			    (cableOrigin.Entity.Layer.editingType == AbstractLayer.EditingType.SourcePoint || cableOrigin.Entity.Layer.editingType == AbstractLayer.EditingType.SourcePolygonPoint))
+				return false;
 
-		// Delete energy_output
-		JObject dataObject = new JObject();
-		dataObject.Add("id", databaseID);
-		batch.AddRequest(Server.DeleteEnergyOutput(), dataObject, BatchRequest.BATCH_GROUP_ENERGY_DELETE);
-	}
+			//Points cannot connect to themselves
+			if (cableOrigin == this)
+				return false;
+
+			//Green energy can't connect to grey
+			if (Entity.GreenEnergy != cableOrigin.Entity.GreenEnergy)
+				return false;
+
+			return true;
+		}
+
+		/// <summary>
+		/// Called on points to see if they can serve as the start point for a cable
+		/// </summary>
+		public bool CanCableStartAtSubEntity(bool greenCable)
+		{
+			//Sockets and generators can only have 1 connection
+			//if ((Entity.Layer.editingType == AbstractLayer.EditingType.Socket || Entity.Layer.editingType == AbstractLayer.EditingType.SourcePoint || Entity.Layer.editingType == AbstractLayer.EditingType.SourcePolygonPoint) && connections.Count > 0)
```

Comment:
Commented code, as always see if it makes sense to leave it there for the future, otherwise remove it! :)